### PR TITLE
Project edit pre-work setup

### PR DESCRIPTION
**Pull request by NickPhura**
# Overview

## This PR contains the following changes

- Add structure for getting project data for edit, and putting project data for update
- Moved a few view/create specific classes/queries to their own files

## This PR contains the following types of changes

- [ ] New feature (change which adds functionality)
- [ ] Enhancement (improvements to existing functionality)
- [ ] Bug fix (change which fixes an issue)
- [ ] Misc cleanup / Refactoring / Documentation
- [ ] Version change

## Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### General

- [ ] I have performed a self-review of my own code

### Code

- [ ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [ ] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [ ] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [ ] My code follows the established style conventions
- [ ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [ ] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [ ] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Screenshots

Please add any relevant UI screenshots, if applicable.

### CODE REVIEW

File: api/src/models/project-create.ts [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+
+const defaultLog = getLogger('models/project-create');
+
+/**
+ * Processes all POST /project request data.
+ *
+ * @export
+ * @class PostProjectObject
+ */
+export class PostProjectObject {
```
Comment (sdevalapurkar) [inline api/src/models/project-create.ts:12]: can we add a test for this class? :)

File: api/src/models/project-create.ts [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+  sampling_conducted: boolean;
+}
+
+/**
+ * Processes POST /project permit data
+ *
+ * @export
+ * @class PostPermitData
+ */
+export class PostPermitData {
```
Comment (sdevalapurkar) [inline api/src/models/project-create.ts:73]: test for this class as well

File: api/src/models/project-create.ts [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+  }
+}
+
+/**
+ * Processes POST /project funding data
+ *
+ * @export
+ * @class PostFundingData
+ */
+export class PostFundingData {
```
Comment (sdevalapurkar) [inline api/src/models/project-create.ts:240]: and this class too

**Review by sdevalapurkar — COMMENTED** [review]
looks good!

can we add tests to missing areas in `models/` and `queries/` files? should be mostly copies of existing tests in other places to ensure the classes set properties correctly in constructor/that the sqlstatement returned is not null when expected :)

**Comment by anissa-agahchen** [comment]
@shreyas, I think we might need to tackle the tests in another PR, so that we can start using this common code base

BUT ... we should have a place to capture the list of tests to be done sooner than later

**Comment by sdevalapurkar** [comment]
> @shreyas, I think we might need to tackle the tests in another PR, so that we can start using this common code base
> 
> BUT ... we should have a place to capture the list of tests to be done sooner than later

hmmm... yeah I'm ok leaving it to another PR but 2.28% is quite a lot and definitely needs to be addressed 
happy to help address it too - but we need to as a team

**Comment by anissa-agahchen** [comment]
> > @shreyas, I think we might need to tackle the tests in another PR, so that we can start using this common code base
> > BUT ... we should have a place to capture the list of tests to be done sooner than later
> 
> hmmm... yeah I'm ok leaving it to another PR but 2.28% is quite a lot and definitely needs to be addressed
> happy to help address it too - but we need to as a team

Yeah, we can help address it better once we have it in dev

**Review by anissa-agahchen — APPROVED** [review]
It looks good in general .. the codecov went down, but I think it's worth putting the PR through and we can collectively address the test coverage

@NickPhura, what approach should go with?

**Comment by NickPhura** [comment]
> 
> 
> It looks good in general .. the codecov went down, but I think it's worth putting the PR through and we can collectively address the test coverage
> 
> @NickPhura, what approach should go with?

I'd rather get this merged now, and include the necessary tests in each of the individual edit ticket PRs.  This is working when I test it with postman, but it hasn't been hooked up the the frontend, so it might be that I've overlooked something.  It would be a bit safer to write tests for each edit workflow after getting its full end-to-end working.

**Comment by NickPhura** [comment]
> 
> 
> > @shreyas, I think we might need to tackle the tests in another PR, so that we can start using this common code base
> > BUT ... we should have a place to capture the list of tests to be done sooner than later
> 
> hmmm... yeah I'm ok leaving it to another PR but 2.28% is quite a lot and definitely needs to be addressed
> happy to help address it too - but we need to as a team

Yeah, we can/should add tests as we flush out each of the edit pieces.  But right now, with this being an incomplete piece of the round-trip workflow, I'd prefer to hold off in case something critical needs to change.

**Pull request merged**